feat(no-await-sync-queries): add auto-fix

neriyarden opened the pull request:
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).

## Changes

<!-- List the changes you're making with this pull request. -->

 - Add auto-fix for the no-await-sync-queries rule (remove 'await ' before sync queries)
 - Update tests
 - Update docs

Do we want to remove the `async` from the test callback when there's no other `await`s in it?

## Context
Fixes an item in #202 
<!--
If you're fixing an issue with this pull request then use the "Fixes" keyword, like this:
Fixes #123
-->

G-Rath reviewed (changes requested):
this looks good but you're assuming there will always be a space after the `await` which means `await(getByIcon('search'))` will error.

I'm pretty sure you can fix this by just removing the space, since the fixer doesn't need to care about the styling of the output

neriyarden commented:
@G-Rath Good catch and thanks for the review 🙂 
I removed the space after the `await` in the fixer and updated the tests.
I also added a test cade for the example you gave (`await(getByIcon('search'))`)

Let me know what you think:)

G-Rath reviewed (approved):
looks good to me!

neriyarden commented:
@Belco90 @MichaelDeBoey
Any update about this PR? Will it be merged?

Belco90 commented:
Hi @neriyarden. I'll try to get this merged soon, thanks for your patience!

Belco90 commented on README.md:
```diff
@@ -299,7 +299,7 @@ module.exports = [
 | [consistent-data-testid](docs/rules/consistent-data-testid.md)                   | Ensures consistent usage of `data-testid`                                                    |                                                                                    |                                                                     |     |
 | [no-await-sync-events](docs/rules/no-await-sync-events.md)                       | Disallow unnecessary `await` for sync events                                                 | ![badge-angular][] ![badge-dom][] ![badge-react][]                                 |                                                                     |     |
-| [no-await-sync-queries](docs/rules/no-await-sync-queries.md)                     | Disallow unnecessary `await` for sync queries                                                | ![badge-angular][] ![badge-dom][] ![badge-marko][] ![badge-react][] ![badge-vue][] |                                                                     |     |
+| [no-await-sync-queries](docs/rules/no-await-sync-queries.md)                     | Disallow unnecessary `await` for sync queries                                                | ![badge-angular][] ![badge-dom][] ![badge-marko][] ![badge-react][] ![badge-vue][] |                                                                     | 🔧  |
```
I think there are some conflicts here, can you resolve that @neriyarde?

neriyarden replied:
Yes, thanks.

Im getting some type errors in these files when im trying to solve the conflicts:
- lib/create-testing-library-rule/index.ts
- lib/rules/await-async-utils.ts
- tools/generate-configs/utils.ts
- lib/utils/types.ts
- lib/utils/compat.ts


I don't remember doing any changes in these files but i still get type errors.

Since I don't know these changes and i don't want to break anythig, can I get some help solving these from someone who's familiar with these files and type?

Belco90 commented:
Closing due to inactivity. Feel free to reopen to get it merged. First, we need to tackle comments from #917
